extensions/wikia/CommunityPage/tests/WikiTopicTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+class WikiTopicTest extends WikiaBaseTest {
+
+	private $communityi18n= [
+		'en' => 'Community',
+		'es' => 'Comunidad',
+		'fr' => 'Communauté',
+		'pl' => 'Społeczność',
+		'ru' => 'Сообщество'
+	];
+
+	public function setUp() {
+		$this->setupFile = __DIR__ . '/../CommunityPage.setup.php';
+		parent::setUp();
+	}
+
+	/**
+	 * @dataProvider fallbackToSitenameProvider
+	 */
+	public function testFallbackToSitename( $sitename, $wikiTopic, $expectedWikiTopic, $lang ) {
+		$this->mockGlobalVariable( 'wgWikiTopic', $wikiTopic );
+		$this->mockGlobalVariable( 'wgSitename', $sitename );
+		$mock = $this->getMock( 'Message', ['plain'] );
+		$mock->expects( $this->any() )->method( 'plain' )->willReturn( $this->communityi18n[$lang] );
+		$this->mockGlobalFunction( 'wfMessage', $mock );
+
+		$this->assertEquals( $expectedWikiTopic, WikiTopic::getWikiTopic() );
+	}
+
+	/**
+	 * @dataProvider preparingWikiTopicFromSitenameProvider
+	 */
+	public function testPreparingWikiTopicFromSitename( $sitename, $expectedWikiTopic, $lang ) {
+		$this->mockGlobalVariable( 'wgWikiTopic', null );
+		$this->mockGlobalVariable( 'wgSitename', $sitename );
+		$mock = $this->getMock( 'Message', ['plain'] );
+		$mock->expects( $this->any() )->method( 'plain' )->willReturn( $this->communityi18n[$lang] );
+		$this->mockGlobalFunction( 'wfMessage', $mock );
+
+		$this->assertEquals( $expectedWikiTopic, WikiTopic::getWikiTopic() );
+	}
+
+	public function fallbackToSitenameProvider() {
+		return [
+			[ 'Wookiepedia', 'Star Wars', 'Star Wars' ],
+			[ 'Wookiepedia', null, 'Wookiepedia' ],
```

Comment:
"Wookiepedia" -> "Wookieepedia" for all of these or Wookieepedians yell at you :P.